### PR TITLE
Review

### PR DESCRIPTION
**Pull request by bethau**

### CODE REVIEW

**Comment by FlorianBagarMicrosoft** [comment]
> ms.author: bethau

Test, are you happy now?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:5 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Holographic remoting allows an app to target a HoloLens with holographic content hosted on a desktop PC or on a UWP device such as the Xbox One, allowing access to more system resources and making it possible to integrate remote [immersive views](app-views.md) into existing desktop PC software. A remoting host app receives an input data stream from a HoloLens, renders content in a virtual immersive view, and streams content frames back to HoloLens. The connection is made using standard Wi-Fi. To use remoting, NuGet package is used to add Holographic Remoting to a desktop or UWP app, and write code to handle the connection and to render in an immersive view.

proposal: Holographic Remoting is added to a desktop or UWP app via a NuGet packet. Additionally, code needs to be added which handles the connection and renders in an immersive view.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:17 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> >[!NOTE]

IMPORTANT?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:27 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> >The **1.x.x** version of the NuGet package is still available for developers that want to target HoloLens 1. For details see [Add Holographic Remoting (HoloLens 1)](holographic-remoting-create-host-hl1.md).

who

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:43 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> >Calling any Windows Mixed Reality related API before creating the remote context can result in unexpected behavior as creating the remote context will start redirecting the API calls. Any calls made before creating the remote context will end up on the local machine. Ensure that the remote context is created before interacting with any Mixed Reality related API. Never mix objects retrieved before the CreateRemoteContext call with objects retrieved after the call.

proposal: Holographic Remoting works by replacing the native Windows Mixed Reality runtime which is part of Windows with a remoting specific runtime. This is done when you create remote context in your app. For that reason any call on the Windows Mixed Reality API before creating the remote context can result in unexpected behavior. The safest approach is to create the remote context as early as possible before interaction with any Mixed Reality API. Also keep in mind to never mix objects created or retrieved before the call to CreateRemoteContext with objects created or retrieved afterwards.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:72 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Next the holographic space needs to be created. Specifying a CoreWindow is not required. Desktop apps can just pass a ```nullptr```.

Remoting apps?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:74 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> A connection from the host app to the device call the ```Connect``` method on the remote context specifying the hostname and port. The port used by the Holographic Remoting Player is 8265.

need to call

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:88 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> A connection from the host app to the device call the ```Connect``` method on the remote context specifying the hostname and port. The port used by the Holographic Remoting Player is 8265.

I would make this bold or colored.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:88 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> A connection from the host app to the device call the ```Connect``` method on the remote context specifying the hostname and port. The port used by the Holographic Remoting Player is 8265.

proposal: To establish a connection from the host app to the HoloLens, the host app needs to call the Connect method on the remote context using the hostname and port of the HoloLens.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:88 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> >[!NOTE]

IMPORTANT?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:101 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 2) OnDisconnected: Triggered if a established connection is separated or a connection could not be established.

an

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:137 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 2) OnDisconnected: Triggered if a established connection is separated or a connection could not be established.

closed?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:137 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Using the remote speech interface it is possible to register speech triggers on the HoloLens 2 device and have them remoted to the host application.

With?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:180 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Using the remote speech interface it is possible to register speech triggers on the HoloLens 2 device and have them remoted to the host application.

just HoloLens 2
I have seen other PR on the docu where the removed things like the device, etc.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:180 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Using an asynchronous helper method we then initialize the remote speech. This should be done asynchronously as initializing might take a considerable amount of time. [Concurrency and asynchronous operations with C++/WinRT](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency) explains how to author asynchronous functions with C++/WinRT.

you can

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:198 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 2) Specification them inside the dictionary vector passed to ```ApplyParameters```.

Specify them via

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:236 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 1) Specification inside a speech grammar xml file. See [How to create a basic XML Grammar](https://docs.microsoft.com/en-us/previous-versions/office/developer/speech-technologies/hh361658(v=office.14)) for details.

Specify by?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:235 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> There are two ways of specifying phrases to be recognized.

to specify

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:234 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> void SampleHostMain::OnRecognizedSpeech(const winrt::hstring& recognizedText)

Should we add the callback registration as well here? You also mention m_onRecognizedSpeechRevoker but do not use it elsewhere in sample code.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:241 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Custom data channels can be used to send user data over the already established remoting connection.

an

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:267 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by bethau** [comment]
> Holographic remoting allows an app to target a HoloLens with holographic content hosted on a desktop PC or on a UWP device such as the Xbox One, allowing access to more system resources and making it possible to integrate remote [immersive views](app-views.md) into existing desktop PC software. A remoting host app receives an input data stream from a HoloLens, renders content in a virtual immersive view, and streams content frames back to HoloLens. The connection is made using standard Wi-Fi. To use remoting, NuGet package is used to add Holographic Remoting to a desktop or UWP app, and write code to handle the connection and to render in an immersive view.

This is actually the exact paragraph from the v1 documentation. Should we still change it?

---
In reply to: [516322262](https://github.com/FlorianBagarMicrosoft/mixed-reality/pull/1#issuecomment-516322262) [](ancestors = 516322262)

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:17 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by bethau** [comment]
> There are two ways of specifying phrases to be recognized.

It would be "There are two ways to specify phrases to be recognized" which is one "to" to many for my taste.

---
In reply to: [516402190](https://github.com/FlorianBagarMicrosoft/mixed-reality/pull/1#issuecomment-516402190) [](ancestors = 516402190)

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:234 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by bethau** [comment]
> void SampleHostMain::OnRecognizedSpeech(const winrt::hstring& recognizedText)

The above code block shows the callback registration. It also shows the type of onRecognizedSpeechRevoker in the function signature. I never use m_onRecognizedSpeechRevoker

---
In reply to: [516402893](https://github.com/FlorianBagarMicrosoft/mixed-reality/pull/1#issuecomment-516402893) [](ancestors = 516402893)

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:241 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> When streaming content to a HoloLens device over a insecure Network such as an open WiFi or the internet it is highly recommended to use an encrypted connection.

to a HoloLens device -> to HoloLens 2

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:17 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 1) A certificate object, which needs to implement the ICertificate interface.

```ICertificate``` ?

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:36 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
>     * Must return the pfx password through ```GetPfxPassword```. Return an empty string for an unprotected pfx.

I'm not sure about the 'Must' at the beginning of each point. Would just start with 'Return' work as well?

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:40 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 2) A certificate provider implementing the ```ICertificateProvider``` which must provide a certificate when asked through the ```GetCertificate``` method.

the ```ICertificateProvider``` which must provide a certificate -> the ```ICertificateProvider``` interface which provides a certificate

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:42 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
>     * ```ValidateCertificate``` is called by the client context to request validation of a certificate. This method accepts the certificate chain (with the first certificate being the subject certificate), the name of the server the connection is being established with, and whether a revocation check should be forced. The system validation result will be provided if validation by the underlying system has been requested. To continue processing, one of the passed ```ICertificateValidationCallback``` methods needs to be called. This can either be ```CertificateValidated``` with the appropriate result or ```Cancel``` to cancel validation.

To continue processing, one of the passed ```ICertificateValidationCallback``` methods needs to be called. This can either be ```CertificateValidated``` with the appropriate result or ```Cancel``` to cancel validation.
->
To continue processing either ```CertificateValidated``` with the appropriate result or ```Cancel``` to cancel validation needs to be called on the passed (in?) ```ICertificateValidationCallback```.

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:46 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Furthermore to allow for the exchange of a secure token the following objects need to be implemented.

Furthermore to -> Furthermore, to

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:48 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> 1) An authentication provider implementing the ```IAuthenticationProvider``` interface. Its ```GetToken``` method is called by the client context to request a token for client authentication. To continue, one of the passed ```IAuthenticationProviderCallback``` methods needs to be called. This can either be ```TokenReceived``` to provide the authentication token and continue the connection process or ```Cancel``` to cancel the process.

To continue, -> To continue either TokenReceived to provide the authentication token and continue the connection process or Cancel to cancel the process needs to be called on the passed (in?) IAuthenticationProviderCallback.

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:50 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
>     * The ```ValidateToken``` method is called by the server network context to request validation of a client authentication token. To continue, one of the passed ```IAuthenticationReceiverCallback```'s methods needs to be called. Either ```ValidationCompleted``` to signal completion of the validation. The client connection will be admitted or rejected based on the passed validation result. Or ```Cancel``` to reject the client connection.

same as with the other two, I would directly start with what methods you need to call to continue.

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:53 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

**Comment by FlorianBagarMicrosoft** [comment]
> Once these objects are implemented ```ListenSecure``` needs to be called instead of ```Listen``` and ```ConnectSecure``` instead of connect on the remote context and player context respectively. ```ListenSecure``` requires an additional certificate provider and authentication receiver over ```Listen```. ```ConnectSecure``` requires an additional authentication provider and certificate validator over ```Connect```.

instead of connect on -> instead of ```Connect``` on

---
Refers to: mixed-reality-docs/holographic-remoting-secure-connection.md:55 in fdc096a. [](commit_id = fdc096a4b06ef8680c731c1c399f06c9edb59919, deletion_comment = False)

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ keywords: HoloLens, Remoting, Holographic Remoting
 ---
 
 
+
 # Writing a Holographic Remoting Host app
 
 By creating a Holographic Remoting host app remote content that is rendered on a remote machine can be streamed to a HoloLens device. This article describes how this can be achieved. All code on this page and working projects can be found in the [Holographic Remoting samples github repository](https://github.com/microsoft/MixedReality-HolographicRemoting-Samples).
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >HoloLens device [](start = 117, length = 15)

HoloLens 2

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 2]
```diff
@@ -9,11 +9,12 @@ keywords: HoloLens, Remoting, Holographic Remoting
...
 
-Holographic remoting allows an app to target a HoloLens with holographic content hosted on a desktop PC or on a UWP device such as the Xbox One, allowing access to more system resources and making it possible to integrate remote [immersive views](app-views.md) into existing desktop PC software. A remoting host app receives an input data stream from a HoloLens, renders content in a virtual immersive view, and streams content frames back to HoloLens. The connection is made using standard Wi-Fi. To use remoting, NuGet package is used to add Holographic Remoting to a desktop or UWP app, and write code to handle the connection and to render in an immersive view.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >HoloLens [](start = 47, length = 8)

HoloLens 2

**Comment by NPohl-MSFT** [comment]
> ## Perquisites

Prerequisites

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:23 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 3]
```diff
@@ -23,7 +24,7 @@ A typical remoting connection will have as low as 50 ms of latency. The player a
 
 A good starting point is a working DirectX based Desktop or UWP app that already targets the Windows Mixed Reality API. For details see [DirectX development overview](directx-development-overview.md). The [C++ holographic project template](creating-a-holographic-directx-project.md) is a good starting point.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >already [](start = 73, length = 7)

remove "already"

**Comment by NPohl-MSFT** [comment]
> ## Get the remoting NuGet packages

## Get the Holographic Remoting NuGet package


---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:32 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

**Comment by NPohl-MSFT** [comment]
> The following steps are required to add the NuGet package to a project in Visual Studio.

...Holographic Remoting...


---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:34 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 4]
```diff
@@ -68,9 +69,9 @@ CreateRemoteContext(m_remoteContext, 20000, false, PreferredVideoCodec::Default)
...
+> Holographic Remoting works by replacing the Windows Mixed Reality runtime which is part of Windows with a remoting specific runtime. This is done during the creation of the remote context. For that reason any call on any Windows Mixed Reality API before creating the remote context can result in unexpected behavior. The recommended approach is to create the remote context as early as possible before interaction with any Mixed Reality API. Never mix objects created or retrieved through any Windows Mixed Reality API before the call to CreateRemoteContext with objects created or retrieved afterwards.
 
-Next the holographic space needs to be created. Specifying a CoreWindow is not required. Desktop apps can just pass a ```nullptr```.
+Next the holographic space needs to be created. Specifying a CoreWindow is not required. Desktop apps that do not own a CoreWindow can just pass a ```nullptr```.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >own [](start = 114, length = 3)

own -> have?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 5]
```diff
@@ -84,7 +85,7 @@ Connection can be done in one of two ways.
 1) The host app connects to the player running on the device.
 2) The player running on the device connects to the host app.
 
-A connection from the host app to the device call the ```Connect``` method on the remote context specifying the hostname and port. The port used by the Holographic Remoting Player is 8265.
+To establish a connection from the host app to the device call the ```Connect``` method on the remote context specifying the hostname and port. The port used by the Holographic Remoting Player is **8265**.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: device -> HoloLens 2?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 6]
```diff
@@ -97,13 +98,13 @@ catch(winrt::hresult_error& e)
 }
 ```
 
->[!NOTE]
+>[!IMPORTANT]
 >As with any C++/WinRT API ```Connect``` might throw an winrt::hresult_error which needs to be handled.
 
 >[!TIP]
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >TIP [](start = 3, length = 3)

Use INFO instead? As this 'tip' hints you something, that is actually NOT recommenced?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 7]
```diff
@@ -97,13 +98,13 @@ catch(winrt::hresult_error& e)
...
 >To avoid using [C++/WinRT](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/) language projection the file ```build\native\include\<windows sdk version>\abi\Microsoft.Holographic.AppRemoting.h``` located inside the Holographic Remoting NuGet package can be included. It contains declarations of the underlying COM interfaces. The use of C++/WinRT is recommended though.
 
-Listening for incoming connections on the host app can be done by calling the ```Listen``` method. Both the handshake port and transport port can be specified during this call. The handshake port is used for the initial handshake. The data is then send over the transport port. By default 8265 and 8266 are used.
+Listening for incoming connections on the host app can be done by calling the ```Listen``` method. Both the handshake port and transport port can be specified during this call. The handshake port is used for the initial handshake. The data is then send over the transport port. By default **8265** and **8266** are used.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >The data is then send over the transport port. [](start = 231, length = 46)

After a successful handshake, the data is sent to the transport port?

**Comment by NPohl-MSFT** [comment]
> 

Move this up to the Get NuGet package section?

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:122 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

**Comment by NPohl-MSFT** [comment]
> The remote context exposes three events which are important to monitor the state of a connection.

The ```RemoteContext``` exposes three events to monitor the state of the connection

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:125 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

**Comment by NPohl-MSFT** [comment]
>             // Failure reason None indicates a normal disconnect.

This comment is wrong. Also it is like this in the sample (where we need to correct it also at some point), we shouldn't copy it to the documentation.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:155 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

**Comment by NPohl-MSFT** [comment]
> First the remote speech interface needs to be retrieved.

First retrieve the remote speech interface.

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:189 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 8]
```diff
@@ -194,7 +195,7 @@ if (auto remoteSpeech = m_remoteContext.GetRemoteSpeech())
 }
 ```
 
-Using an asynchronous helper method we then initialize the remote speech. This should be done asynchronously as initializing might take a considerable amount of time. [Concurrency and asynchronous operations with C++/WinRT](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency) explains how to author asynchronous functions with C++/WinRT.
+Using an asynchronous helper method you can then initialize the remote speech. This should be done asynchronously as initializing might take a considerable amount of time. [Concurrency and asynchronous operations with C++/WinRT](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency) explains how to author asynchronous functions with C++/WinRT.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >Using an asynchronous helper method you can then initialize the remote speech [](start = 0, length = 77)

Initialize the remote speech by using an asynchronous helper method.

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 9]
```diff
@@ -232,7 +233,7 @@ winrt::fire_and_forget InitializeSpeechAsync(
 
 There are two ways of specifying phrases to be recognized.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: Phrases to be recognized can be specified in two ways?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 10]
```diff
@@ -232,7 +233,7 @@ winrt::fire_and_forget InitializeSpeechAsync(
...
 1) Specification inside a speech grammar xml file. See [How to create a basic XML Grammar](https://docs.microsoft.com/en-us/previous-versions/office/developer/speech-technologies/hh361658(v=office.14)) for details.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >Specification inside [](start = 3, length = 20)

Remove "Specification inside"?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 11]
```diff
@@ -232,7 +233,7 @@ winrt::fire_and_forget InitializeSpeechAsync(
...
-2) Specification them inside the dictionary vector passed to ```ApplyParameters```.
+2) Specify by passing them inside the dictionary vector to ```ApplyParameters```.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >Specify by passing them inside the dictionary vector to ```ApplyParameters```. [](start = 3, length = 78)

Passing a dictionary vector to ```ApplyParameters```.?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 12]
```diff
@@ -232,7 +233,7 @@ winrt::fire_and_forget InitializeSpeechAsync(
...
 
 Inside the OnRecognizedSpeech callback the speech events can then be processed:
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >the [](start = 39, length = 3)

remove the second "the"?

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 13]
```diff
@@ -261,69 +262,6 @@ void SampleHostMain::OnRecognizedSpeech(const winrt::hstring& recognizedText)
...
-{
-    std::lock_guard lock(m_customDataChannelLock);
-    if (m_customDataChannel)
-    {
-        m_customChannelDataReceivedEventRevoker.revoke();
-        m_customChannelClosedEventRevoker.revoke();
-        m_customDataChannel = nullptr;
-    }
-}
-```
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: Add a short paragraph with a link to the documentation instead?
E.g.


Optional: Custom data channels

Custom data channels can be used to send user data over the already established remoting connection. See [custom data channels](holographic-remoting-custom-data-channels.md) for more information.

File: mixed-reality-docs/holographic-remoting-create-host.md [hunk 14]
```diff
@@ -261,69 +262,6 @@ void SampleHostMain::OnRecognizedSpeech(const winrt::hstring& recognizedText)
...
-
 ## Preview streamed content locally
 
 To display the same content in the host app that is send to the device the ```OnSendFrame``` event of the remote context can be used. The ```OnSendFrame``` event is triggered every time the Holographic Remoting library sends the current frame to the remote device. This is the ideal time to take the content and also blit it into the desktop or UWP window.
```
Comment (NPohl-MSFT) [inline mixed-reality-docs/holographic-remoting-create-host.md]: >The ```OnSendFrame``` event is triggered every time the Holographic Remoting library sends the current frame to the remote device [](start = 134, length = 129)

Should we be more elaborate when this does actually happen? Also: Should we document that it is the content which is rendered to the BackBuffer of the HolographicFrame which is actually transfered?

**Comment by bethau** [comment]
> 

The intention here is to point the user to the idl as documentation as we are using a method from that idl the first time here. In my opinion this would be completely misplaced in the NuGet section.

---
In reply to: [516503603](https://github.com/FlorianBagarMicrosoft/mixed-reality/pull/1#issuecomment-516503603) [](ancestors = 516503603)

---
Refers to: mixed-reality-docs/holographic-remoting-create-host.md:122 in 5382504. [](commit_id = 53825048ca62aa573dc0349a6610066ad00ed3bb, deletion_comment = False)

**Pull request merged**